src/include/stir/FilePath.h

```diff
@@ -0,0 +1,170 @@
+/*
+    Copyright (C) 2018, University of Hull
+
+    This file is part of STIR.
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+
+#ifndef __stir_FILEPATH_H__
+#define __stir_FILEPATH_H__
+
+/*!
+  \file
+  \ingroup buildblock
+
+  \brief Declaration of class stir::FilePath
+  This is a class implementing basic filesytem functionality. Parts of this class were
+  copied from Functions for filename manipulations.
+
+  Maximum effort was put in order to be cross platform.
+
+  \author Nikos Efthimiou
+
+*/
+
+#include "stir/common.h"
+#include <stdio.h>
+#include <iostream>
+#include <string>
+#include <boost/format.hpp>
+// N.E: Not sure I want this here. Included for the max_filename_length.
+#include "stir/utilities.h"
```

Comment:
I think you only want this in the .cxx. Probably also the case for the includes below.

Reply:
ok